lahja/asyncio/endpoint.py

```diff
@@ -388,7 +388,11 @@ def _throw_if_already_connected(self, *endpoints: ConnectionConfig) -> None:
         seen: Set[str] = set()
 
         for config in endpoints:
-            if config.name in seen:
+            if config.name == self.name:
+                raise ConnectionAttemptRejected(
+                    f"Trying to connect to {config.name} twice. Names must be uniqe."
```

Comment:
This should probably be a different message. E.g. "Can not connect to own endpoint"